### PR TITLE
[1pt] PR: SRC optimization code refinements

### PR DESCRIPTION
**Pull request by RyanSpies-NOAA**
This feature branch includes minor enhancements to the SRC adjustment code including a new attribute appended to the hydroID catchments .gpkg and an optional input argument to override the default downstream distance threshold for roughness application. Addresses #505 

## Changes

- `tools/adjust_rc_with_feedback.py`: Added new attribute "src_calibrated" to hydroid catchments .gpkg layer, added new input argument (optional `-m` flag) to override the downstream threshold distance for roughness application, fix for `to_crs` to override current crs of input points layer to match wbd layer , updated/cleaned docstrings, and added additional content to the log file
- `tools/rating_curve_comparison.py`: Removed redundant code lines for plotting
- `tools/usgs_gage_src_adjust.py`: Added the USGS gage location id number to the "submitter" attribute stored in the database

## Testing
Tested for both HUC6 and HUC8 runs with `adjust_rc_with_feedback.py`
HUC6 test usage: `python foss_fim/tools/adjust_rc_with_feedback.py -p data/temp/water_edge_rc_adjust/known_water_edge_test_12030102.gpkg -d data/outputs/ryan_temp/aggregate_fim_outputs_test/ -w data/temp/water_edge_rc_adjust/WBD_National_copy.gpkg -i True -s HUC6`

HUC8 test usage: `python3 foss_fim/tools/usgs_gage_src_adjust.py -fim_dir data/outputs/dev_v3_0_24_8_fim_run_ms/ -usgs_rc data/inputs/usgs_gages/usgs_rating_curves.csv -nwm_recur data/inundation_review/inundation_nwm_recurr/nwm_recurr_flow_data/nwm21_17C_recurrence_flows_cfs.csv -i True -j 9`

## Screenshots
Example of the new gw_catchments .gpkg layer with the `src_calibrated` attribute that identifies catchments where a new roughness value has been applied to the synthetic rating curve via `adjust_rc_with_feedback.py`.
image

Example output showing the use of the new `-m` flag in `adjust_rc_with_feeback.py` - note the reduced number of hydroID catchments (blue) that have been altered downstream of the obs point data. Note: by default the downstream distance threshold is set in `tools_shared_variables.py`
image

## Todos

- Update changelog before merging

### CODE REVIEW

**Comment by BradfordBates-NOAA** [comment]
@RyanSpies-NOAA Changelog?

File: tools/adjust_rc_with_feedback.py [hunk 1]
```diff
@@ -165,60 +165,71 @@ def update_rating_curve(fim_directory, water_edge_median_ds, htable_path, output
...
+                output_catchments.to_file(catchments_poly_path,driver="GPKG",index=False) # overwrite the previous layer
+                df_nmerge.drop(['src_calibrated'], axis=1, inplace=True)
+            ## Optional ouputs: 1) merge_n_csv csv with all of the calculated n values and 2) a catchments .gpkg with new joined attributes
+            if optional_outputs == 'True' and os.path.isfile(catchments_poly_path):
                 output_merge_n_csv = os.path.join(fim_directory, huc, 'merge_src_n_vals_' + huc + '.csv')
                 df_nmerge.to_csv(output_merge_n_csv,index=False)
-                # output new catchments polygon layer with the new manning's n value attributes appended
-                if catchments_poly_path != '':
+                ## output new catchments polygon layer with several new attributes appended
+                if os.path.isfile(catchments_poly_path):
```
Comment (CarsonPruitt-NOAA) [inline tools/adjust_rc_with_feedback.py:205]: if statement might be redundant due to same check added on line 201

**Review by CarsonPruitt-NOAA — DISMISSED** [review]
Tested on a subset of MS HUC8's

File: tools/adjust_rc_with_feedback.py (hunk 1 above)
Comment (RyanSpies-NOAA) [reply tools/adjust_rc_with_feedback.py:205]: Good find. I have addressed this.

**Review by BradfordBates-NOAA — APPROVED** [review]

**Pull request merged**